cmd/bootkube/render.go

```diff
@@ -55,6 +59,9 @@ func init() {
 	cmdRender.Flags().StringVar(&renderOpts.assetDir, "asset-dir", "", "Output path for rendered assets")
 	cmdRender.Flags().StringVar(&renderOpts.caCertificatePath, "ca-certificate-path", "", "Path to an existing PEM encoded CA. If provided, TLS assets will be generated using this certificate authority.")
 	cmdRender.Flags().StringVar(&renderOpts.caPrivateKeyPath, "ca-private-key-path", "", "Path to an existing Certificate Authority RSA private key. Required if --ca-certificate is set.")
+	cmdRender.Flags().StringVar(&renderOpts.etcdCAPath, "etcd-ca-path", "", "Path to an existing PEM encoded CA that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-certificate-path and --etcd-private-key-path, and must have etcd configured to use TLS with matching secrets.")
+	cmdRender.Flags().StringVar(&renderOpts.etcdCertificatePath, "etcd-certificate-path", "", "Path to an existing certificate that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-ca-path and --etcd-private-key-path, and must have etcd configured to use TLS with matching secrets.")
```

Comment:
Allowing this makes the use of these three flags (ca, cert, key) have varying behaviors. E.g. if `etcd-certificate-path` is provided, it's assumed that the key is for that cert, not the CA. But if it is not provided, the key is for the CA not the cert.

I'm thinking we could even remove this flag and only allow the option of providing a CA/key to use while generating certs for the etcd nodes. Only allowing the ca cert/key to be provided is more inline with how we generate the rest of the certs (user provides CA, which we use). Providing your own cert is something that is very easy to get wrong as well (wrong CN/SAN, etc.) which is why we've gone the route of "provide CA, but we'll generate).

Also, along those lines I'm wondering if we need to allow for providing a different etcd-specific CA altogether. The flexibility can't hurt -- just thinking aloud.

What about if we changed this to just allow:

```
--etcd-ca-certificate-path=
--etcd-ca-key-path=
```

Then use those to generate the assets we need.

Or, now that we're assuming cert generation based on etcd server scheme -- if we didn't care about supporting a separate etcd-specific CA - we wouldn't really need any new flags (just use the existing `--ca-certificate-x-path` flags).

Sorry for the back-forth on this. Obviously more considerations than I initially thought.

/cc @dghubble @ericchiang any thoughts here? Specifically, do we want a diff CA for etcd? Is it safe to assume that given a CA we can generate the assets in the default case?